src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -812,101 +731,204 @@ public int getCurrentRunningCount() {
+                } else if (nextTask.canExecute(executeReference)) {
+                  return nextTask;
+                }
+              }
+              // reset interrupted status if we may block and have not checked
+              if (queued && ! interruptedChecked) {
+                interruptedChecked = true;
+                if (Thread.interrupted()) {
+                  // verify we were not interrupted due to pool shutdown
+                  continue idle;
+                }
+              }
+            } // end pollTask loop
+          }
+        } // end idle loop
+      } finally {
+        // if queued, we must now remove ourselves, since we are either shutdown, or have a task to run
+        if (queued) {
+          // only if we are the head of the queue can we remove ourselves without locking
+          if (idleWorker.get() == worker && 
+              idleWorker.compareAndSet(worker, worker.nextIdleWorker)) {
+            // no further actions needed, change occurred in if condition
+          } else {
+            /* I wish we could remove this lock...
+             * Right now we must lock because we must atomically update the worker holding our 
+             * reference to point to the worker held inside our reference.  This means two 
+             * references must be updated in a single operation
+             */
+            /* Locking operations here don't interfere with concurrent operations on idleWorker's 
+             * atomic reference (which is why that was checked first)
+             */
+            synchronized (idleWorker) {
+              while (true) {
+                Worker holdingWorker = idleWorker.get();
+                if (holdingWorker == worker) {
+                  if (idleWorker.compareAndSet(worker, worker.nextIdleWorker)) {
+                    break;
+                  } else {
+                    continue;
+                  }
+                }
+                while (holdingWorker != null && holdingWorker.nextIdleWorker != worker) {
+                  // In theory we know we are in the queue, so unless there is a bug
+                  // there should be no reason to check for nulls. 
+                  holdingWorker = holdingWorker.nextIdleWorker;
+                }
+                
+                if (holdingWorker != null) {
+                  // now remove this worker from the chain
+                  holdingWorker.nextIdleWorker = worker.nextIdleWorker;
+                  break;
+                }
+              }
+            }
+          }
+          // now out of the queue, lets clean up our reference
+          worker.nextIdleWorker = null;
+        }
+        
+        // wake up next worker so he can check if tasks are ready to consume
+        handleQueueUpdate();
+        
+        if (! interruptedChecked) {
+          // reset interrupted status
+          Thread.interrupted();
+        }
+      }
+    }
+
+    @Override
+    public void handleQueueUpdate() {
```

Comment:
This is used any time where the pool size has increased, or we want to wake up or start a thread (depending on the pool state).

Though as I type this now, we default to this when we have consumed tasks, which means we may start one more thread than necessary for a given work load.  But for code simplicity it seems worth it.